peer/brontide.go

```diff
@@ -2569,11 +2556,18 @@ func (p *Brontide) handleLocalCloseReq(req *htlcswitch.ChanClose) {
 			return
 		}
 
+		// Optimistically try a link shutdown, erroring out if it
+		// failed.
+		if err := p.tryLinkShutdown(chanID); err != nil {
```

Comment:
Nice catch w/ the addition here as well! Could end up in a weird state triggered by the caller itself.